libs/wingc/src/type_check.rs

```diff
@@ -554,9 +555,13 @@ impl Types {
 		types.push(Box::new(Type::Void));
 		let void_idx = types.len() - 1;
 
+		let void_ref = UnsafeRef::<Type>(&*types[void_idx] as *const Type);
+		let libraries = SymbolEnv::new(None, void_ref, false, false, Phase::Preflight, 0);
```

Comment:
This "hack" for setting the void return value of the env kind of shows why this doesn't fit perfectly into a `SymbolEnv`.

It might be wroth attempting your suggestion of turning our `SymbolEnv` into a trait. And then adding the extra info (`return_type`, `is_init`, `is_class`, 'statement_idx`) to concrete implementations. Another challenge here is to see if we can model different enums of `SymbolKind` for each concrete implementation. 

I can currently think of the following concrete types/implementations of a `SymbolEnv` in our code:
* `CodeEnv`: A `SymbolEnv` associated with a `Scope` - has `return_type`, `is_init`, `statement_idx` fields. Contains Types, Vars, Namespaces.
* `ClassEnv`: A `SymbolEnv` associated with a class/resource/struct type (maybe enums too). Contains Vars only.
* `LibraryEnv`: A `SymbolEnv` associated with imported libraries. Contains Types and Namespaces only.

Because this seems complex I'm fine with what you did here, just add a comment that you can't call `self.void()` from withing the static `new`, and this is just a hack to create the top level env for mapping lib names to Symbols.

Reply:
Added